WeakAuras/SubscribableObject.lua

```diff
@@ -50,7 +50,10 @@ local SubscribableObject =
   --- @type fun(self: SubscribableObject, event: string, subscriber: frame)
   RemoveSubscriber = function(self, event, subscriber)
     if self.events[event] then
-      tremove(self.events[event], tIndexOf(self.events[event], subscriber))
+      local index = tIndexOf(self.events[event], subscriber)
+      if index then
+        tremove(self.events[event], index)
+      end
       if not TableHasAnyEntries(self.events[event]) then
```

Comment:
it wasn't really clear to me if this callback check should also be behind the nil check

Reply:
Yes it should. The callbacks are called on first to subscribe, last o unsubscribe. And writing that I should have named them differently.